### DISCUSSION TITLE
Piped Proxy Downloads

### DISCUSSION THREAD

**Discussion by matt873** [deniscerri/ytdlnis#312] [General]
Libretube can directly download videos through piped servers in the app. But sharing the same video piped URL to YTDLnis downloads through YouTube servers instead of the piped proxy. Is this an issue of yt-dlp itself or YTDLnis is missing this function?

**Comment by deniscerri** [comment]
No its not an issue. What is currently happening is that the app transforms the piped url to a youtube one so it knows what it is then passes it over to piped or yt-dlp as fallback to get data. You think its better to use the proxy rather than doing that?

**Reply by deniscerri** [reply]
But at the end of the day all the app is doing is grabbing the video id and asking piped to get the data anyway. What you might be referring to is the youtube url in the bottom. Thats the transformed url

**Reply by matt873** [reply]
I checked through the pcapdroid app and there were active connections to YouTube servers while downloading through piped URL. Since the purpose of piped is to avoid revealing IP addresse to YouTube servers and hence the reason for Libretube's existence, can't yt-dlp or YTDLnis avoid any connections to YouTube for piped URLs? The purpose of opening this discussion is to make YTDLnis a useful alternative instead of in-app downloads for Libretube users specifically and yt-dlp users in general.
![Screenshot_20231030-011512](https://github.com/deniscerri/ytdlnis/assets/108935493/c6f87900-83c2-4d85-ab0a-110cf8a320dc)
![Screenshot_20231030-011520](https://github.com/deniscerri/ytdlnis/assets/108935493/14ba33dc-d540-4cfd-9f46-55fb8878530f)

**Reply by deniscerri** [reply]
The only way this could happen is if the piped api server failed to return data, so the app falls back to yt-dlp and that requests the youtube servers. Try changing your piped api server url in the updating settings

If it takes a few seconds to load a link for example, thats a good hint that the piped api failed and the app switched to yt-dlp. Piped is almost instant.

**Reply by deniscerri** [reply]
But at the end of the day you will have to use YouTube servers when you download because that's what ytdlp does. If they support piped proxy let me know
